andela/ah-bird-box
# #163388249 verification of email upon signup

#### What does this PR do?
Have verification email upon signup registration
#### Description of Task to be completed?
- add tests for verification
- add email template
- add email settings
- edit jwt tests
#### How should this be manually tested?
* clone the repo and `cd` into it.
* checkout to the branch
  `git checkout ft-email-verification-163388249`

* Create a .env file and add the following.
```
export EMAIL_SENDER="your email"
export EMAIL_HOST="smtp.gmail.com"
export EMAIL_HOST_USER="your email"
export EMAIL_HOST_PASSWORD="your email password"
export EMAIL_PORT=587
```
* Run server
  `python manage.py runserver`
* test on postman using the url `http://127.0.0.1:8000/api/users/`
* On the payload, input username, email and password i.e

```
         {
		"username": "gracey",
		"email": "meffdsfgfdf@gmail.com",
		"password": "Sanjihydfsvs$$$#34!"
	}
```
* Message should appear i.e `User registered successfully. Check your email to activate your account.`

* An email is received once registration is successfull. Click on the link provided to verify email.

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388249](https://www.pivotaltracker.com/n/projects/2240063)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/42655718/52472037-67a45f80-2ba3-11e9-8461-96817a7d18f1.png)

#### Questions:
